clang-tools-extra/clang-tidy/ClangTidyDiagnosticConsumer.h

```diff
@@ -326,6 +331,7 @@ class ClangTidyDiagnosticConsumer : public DiagnosticConsumer {
   bool LastErrorRelatesToUserCode = false;
   bool LastErrorPassesLineFilter = false;
   bool LastErrorWasIgnored = false;
+  bool InSourceFile = false;
```

Comment:
nitpick: maybe add a comment here saying that this flag indicates if the `ClangTidyDiagnosticConsumer` is in between a pair of `BeginSourceFile` and `EndSourceFile` calls.